src/main/java/games/strategy/engine/framework/startup/login/ClientLoginValidator.java

```diff
@@ -98,8 +98,9 @@ public String verifyConnection(
 
     // check for version
     final Version clientVersion = new Version(versionString);
-    if (!ClientContext.engineVersion().equals(clientVersion)) {
-      return "Client is using " + clientVersion + " but server requires version " + ClientContext.engineVersion();
+    if (!ClientContext.engineVersion().isCompatible(clientVersion)) {
+      return "Client is using " + clientVersion + " but server requires version " + ClientContext.engineVersion()
```

Comment:
A space is needed before `"or any` to prevent the engine version number from being jammed up against the word. While here, I recommend updating the wording as well slightly, something like:
```
String.format("Client is using %s but the server requires a version compatible with version %s ",
   clientVersion, ClientContext.engineVersion());
```

Reply:
Confused by these comments. There is a space around all the version numbers.

Reply:
@simon33-2 I think @DanVanAtta is talking about the following line (103).  The string literal needs to start with a space:

```
          + " or any other compatible version";
             ^
             this space is missing
```

Reply:
I already replaced this line, so no worries ^^

Reply:
ok, thanks.